tests/e2e/vcp_to_csi_syncer.go

```diff
@@ -0,0 +1,582 @@
+	})
+
+})
+
+func waitForCnsVSphereVolumeMigrationCrd(ctx context.Context, vpath string) (*v1alpha1.CnsVSphereVolumeMigration, error) {
+	var (
+		found bool
+		crd   *v1alpha1.CnsVSphereVolumeMigration
+	)
+	waitErr := wait.PollImmediate(poll, pollTimeout, func() (bool, error) {
+		found, crd = getCnsVSphereVolumeMigrationCrd(ctx, vpath)
+		return found, nil
+	})
+	return crd, waitErr
+}
+
+func createDir(path string, host string) error {
+	sshCmd := fmt.Sprintf("mkdir -p %s", path)
+	framework.Logf("Invoking command '%v' on ESX host %v", sshCmd, host)
+	result, err := fssh.SSH(sshCmd, host+":22", framework.TestContext.Provider)
+	if err != nil || result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: '%s' on ESX host: %v", sshCmd, err)
+	}
+	return nil
+}
+
+func createVmdk(host string, size string, objType string, diskFormat string) (string, error) {
+	dsName := GetAndExpectStringEnvVar(envSharedDatastoreName)
+	dir := "/vmfs/volumes/" + dsName + "/e2e"
+	err := createDir(dir, host)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	if diskFormat == "" {
+		diskFormat = "thin"
+	}
+	if objType == "" {
+		objType = "vsan"
+	}
+	if size == "" {
+		size = "2g"
+	}
+	rand.Seed(time.Now().UnixNano())
+	vmdkPath := fmt.Sprintf("%s/test-%v-%v.vmdk", dir, time.Now().UnixNano(), rand.Intn(1000))
+	sshCmd := fmt.Sprintf("vmkfstools -c %s -d %s -W %s %s", size, diskFormat, objType, vmdkPath)
+	framework.Logf("Invoking command '%v' on ESX host %v", sshCmd, host)
+	result, err := fssh.SSH(sshCmd, host+":22", framework.TestContext.Provider)
+	if err != nil || result.Code != 0 {
+		fssh.LogResult(result)
+		return vmdkPath, fmt.Errorf("couldn't execute command: '%s' on ESX host: %v", sshCmd, err)
+	}
+	return vmdkPath, nil
+}
+
+func deleteVmdk(host string, vmdkPath string) error {
+	sshCmd := fmt.Sprintf("rm -f %s", vmdkPath)
+	framework.Logf("Invoking command '%v' on ESX host %v", sshCmd, host)
+	result, err := fssh.SSH(sshCmd, host+":22", framework.TestContext.Provider)
+	if err != nil || result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: '%s' on ESX host: %v", sshCmd, err)
+	}
+	return nil
+}
+
+func getCanonicalPath(vmdkPath string) string {
+	dsName := GetAndExpectStringEnvVar(envSharedDatastoreName)
+	parts := strings.Split(vmdkPath, "/")
+	vmDiskPath := "[" + dsName + "] " + parts[len(parts)-2] + "/" + parts[len(parts)-1]
+	datastorePathObj := new(object.DatastorePath)
+	isSuccess := datastorePathObj.FromString(vmDiskPath)
+	gomega.Expect(isSuccess).To(gomega.BeTrue())
+	newParts := strings.Split(datastorePathObj.Path, "/")
+	return strings.Replace(vmDiskPath, parts[len(parts)-2], newParts[len(newParts)-2], 1)
+}
+
+func verifyCnsVolumeMetadataAndCnsVSphereVolumeMigrationCrdForPvcs(ctx context.Context, client clientset.Interface, namespace string, pvcs []*v1.PersistentVolumeClaim) {
```

Comment:
Add  function description on all the functions

Reply:
Done